src/tests/electron/tests/android-setup-multiple-devices.test.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import {
+    leftFooterButtonAutomationId,
+    rightFooterButtonAutomationId,
+} from 'electron/views/device-connect-view/components/android-setup/android-setup-step-layout';
+import { deviceDescriptionAutomationId } from 'electron/views/device-connect-view/components/android-setup/device-description';
+import { getAutomationIdSelector } from 'tests/common/get-automation-id-selector';
+import { createApplication } from 'tests/electron/common/create-application';
+import { scanForAccessibilityIssues } from 'tests/electron/common/scan-for-accessibility-issues';
+import { AndroidSetupViewController } from 'tests/electron/common/view-controllers/android-setup-view-controller';
+import { AppController } from 'tests/electron/common/view-controllers/app-controller';
+import { commonAdbConfigs, setupMockAdb } from '../../miscellaneous/mock-adb/setup-mock-adb';
+
+describe('Android setup - multiple devices ', () => {
+    let app: AppController;
+    let dialog: AndroidSetupViewController;
+
+    beforeEach(async () => {
+        await setupMockAdb(commonAdbConfigs['multiple-devices']);
+        app = await createApplication({ suppressFirstTimeDialog: true });
+        dialog = await app.openAndroidSetupView('prompt-choose-device');
+    });
+
+    afterEach(async () => {
+        if (app != null) {
+            await app.stop();
+        }
+    });
+
+    it('initial component state is correct', async () => {
+        const [closeId, nextId] = [leftFooterButtonAutomationId, rightFooterButtonAutomationId];
+        expect(await dialog.isEnabled(getAutomationIdSelector(closeId))).toBe(true);
+        expect(await dialog.isEnabled(getAutomationIdSelector(nextId))).toBe(false);
+        expect(await dialog.isEnabled(getAutomationIdSelector('rescan'))).toBe(true);
+        const devices = await dialog.client.$$(
+            getAutomationIdSelector(deviceDescriptionAutomationId),
+        );
+        expect(devices.length).toBe(3);
+    });
+
+    it('choosing a device changes next button state, selecting next goes to detect-service', async () => {
```

Comment:
Should we consider a test that ensures that the selected device is properly chosen in  future steps? Possibly mock a failure while installing the service then check the device? Just a thought...

Reply:
I think it's an interesting idea, I wonder if the mock-adb command would fail if the same device weren't requested later

Reply:
`workingDeviceComands`, which builds the config file, enumerates through all 3 devices to ensure that they would all work. That being said, if the test were to start with `multiple-devices`, then overlay it with responses to cause mock-adb to throw for the devices that aren't being selected, then we could distinguish between the 2 cases. If we end up in detect-permissions, then it used the correct device. If we end up in prompt-install-service, then we know that the correct device was not used.